### PR TITLE
fix: use current location as base url for form action

### PR DESCRIPTION
**Pull request by itsMapleLeaf**
Attempting to use the `action` attribute with `Form` raises an error. The `new URL` constructor also needs a base URL if just being passed a path, which serves as the "fallback" of whatever's incomplete in the first arg. So `new URL('/somepath', window.location.href)` should come out to `https://localhost:4000/somepath` (assuming localhost 4k)

Also includes some generic type fixes and improvements

### CODE REVIEW

File: src/form.tsx [hunk 1]
```diff
@@ -86,7 +89,7 @@ async function fetchData({
 }: FetchDataOptions): Promise<Response> {
   // patch must be in all caps: https://github.com/github/fetch/issues/254
   const method = methodArg.toUpperCase();
-  const url = new URL(urlArg);
+  const url = new URL(urlArg, window.location.href);
```
Comment (smeijer) [inline src/form.tsx:93]: Should we only do this if urlArg is relative? What if someone provides the full url as action?

Comment (itsMapleLeaf) [reply src/form.tsx:93]: If someone passes a full URL, it replaces the window location:

image

File: package.json [hunk 2]
```diff
@@ -43,15 +47,7 @@
...
     "@testing-library/user-event": "^13.2.1",
-    "@types/accepts": "^1.3.5",
-    "@types/busboy": "^0.2.4",
-    "@types/bytes": "^3.1.1",
-    "@types/cookies": "^0.7.7",
-    "@types/debug": "^4.1.7",
     "@types/jest": "^27.0.1",
-    "@types/node-fetch": "^2.5.12",
-    "@types/react": "^17.0.20",
-    "@types/test-listen": "^1.1.0",
```
Comment (smeijer) [inline package.json:54]: I think something went wrong with a rebase?

Comment (itsMapleLeaf) [reply package.json:54]: Oops 😂 I'll fix right quick

**Comment by smeijer** [comment]
published https://github.com/smeijer/next-runtime/releases/tag/v2.0.2

**Pull request merged**